agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -348,6 +410,24 @@ func (cs *CredentialSpecResource) clearCredentialSpec() {
 	}
 }
 
+func (cs *CredentialSpecResource) deleteTaskExecutionCredentialsRegKeys() {
+	cs.lock.Lock()
+	defer cs.lock.Unlock()
+
+	pathToTaskCredentialsRegistryKey := servicesRegistryKeyRoot + "\\" + taskCredentialsRegistryKey
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, pathToTaskCredentialsRegistryKey, registry.ALL_ACCESS)
+	if err != nil {
+		seelog.Errorf("Error opening %s key to delete: %s", pathToTaskCredentialsRegistryKey, err)
```

Comment:
We should handle not found and not call it an error. Thats a success early exit

Reply:
Also line 424 will surely fail if open fails. So I think we need to early exit here

Reply:
done